lupa/__init__.py

```diff
@@ -52,7 +50,15 @@ def _import_newest_lib():
         raise RuntimeError("Failed to import Lupa binary module.")
     # prefer Lua over LuaJIT and high versions over low versions.
     module_name = max(modules, key=lambda m: (m[1] == 'lua', tuple(map(int, m[2] or '0'))))
-    _newest_lib = __import__(module_name[0], level=1, fromlist="*", globals=globals())
+
+    # Allowing module loading using dlopenflags by default doesn't work when there are multiple
+    # Lua modules because the symbols collide with each other when loaded with RTLD_GLOBAL.
+    # Enable this by default only if there is exactly one lua module available.
+    if len(modules) == 1:
```

Comment:
Hmm. I would like to have something clear and predictable. This seems a likely surprise for users. Loading Lua modules through exactly the same code will work on some systems but not everywhere.

Reply:
Given the way that `setup.py` works today this will be true if and only if the install is using system Lua,
The system setup only finds one preferred version and bundled always includes at least all of the regular Lua versions.

I'm totally OK if you just want to drop the last commit off this PR and just push this problem onto the users. As you said I'm probably an atypical user wanting to load modules in my Lua.